### PR TITLE
Fix StatefulSet created should allow to scale up after scale down to zero test

### LINKED ISSUE #3546: [Flaky e2e] StatefulSet integration when StatefulSet created should allow to scale up after scale down to zero.
/kind flake

**What happened**:
End To End Suite: kindest/node:v1.30.0: [It] StatefulSet integration when StatefulSet created should allow to scale up after scale down to zero

**What you expected to happen**:
No random failures

**How to reproduce it (as minimally and precisely as possible)**:
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/3539/pull-kueue-test-e2e-main-1-30/1857384106699001856

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug
/kind failing-test
/kind flake

#### What this PR does / why we need it:
Fix StatefulSet created should allow to scale up after scale down to zero test

#### Which issue(s) this PR fixes:

Fixes #3546

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mbobrovskyi** [comment]
/test pull-kueue-test-e2e-main-1-28 pull-kueue-test-e2e-main-1-29 pull-kueue-test-e2e-main-1-30 pull-kueue-test-e2e-main-1-31

2 of 5

**Comment by mbobrovskyi** [comment]
/test pull-kueue-test-e2e-main-1-28 pull-kueue-test-e2e-main-1-29 pull-kueue-test-e2e-main-1-30 pull-kueue-test-e2e-main-1-31

3 of 5

File: test/e2e/singlecluster/statefulset_test.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ var _ = ginkgo.Describe("StatefulSet integration", func() {
 					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(statefulSet), createdStatefulSet)).To(gomega.Succeed())
 					createdStatefulSet.Spec.Replicas = ptr.To[int32](3)
 					g.Expect(k8sClient.Update(ctx, createdStatefulSet)).To(gomega.Succeed())
-				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply test/e2e/singlecluster/statefulset_test.go]: Done

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**